cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -176,19 +177,65 @@ mkTracers traceOptions tracer = Tracers
         Tracer $ \(WithSeverity _ (WithTip _tip ev')) ->
           case ev' of
               (ChainDB.TraceAddBlockEvent ev) -> case ev of
-                  ChainDB.SwitchedToChain _ c -> do
-                    case withOriginToMaybe . pointSlot $ AF.headPoint c of
-                      Nothing -> pure () -- We are at genesis block.
-                      Just slotNum -> do let sNum :: Integer
-                                             sNum = fromIntegral $ unSlotNo slotNum
-                                             logValue :: LOContent a
-                                             logValue = LogValue "slotNum" . PureI $ sNum
-                                         -- phash = fst tippair
-                                         meta <- mkLOMeta Debug Confidential
-                                         let tr' = appendName "slotNum" tr
-                                         traceNamedObject tr' (meta, logValue)
+                  ChainDB.SwitchedToChain _ c ->
+                    when (slots c > 0) $ do
+                      meta <- mkLOMeta Notice Public
+                      let tr' = appendName "metrics" tr
+                          sNum :: Integer = fromIntegral $ slotN c
+                          bNum :: Integer = fromIntegral $ blockN c
+                          sl = slots c
+                          bl = blocks c
+                          density = toRational bl / toRational sl
+                          epochSlots :: Integer = 21600  -- TODO
+                      traceNamedObject tr (meta, LogMessage $ "density based on blocks:" <> pack (show bl) <> " slots:" <> pack (show sl))
+                      traceNamedObject tr' (meta, LogValue "density" . PureD $ fromRational density)
+                      traceNamedObject tr' (meta, LogValue "slotNum" . PureI $ sNum)
+                      traceNamedObject tr' (meta, LogValue "blockNum" . PureI $ bNum)
+                      traceNamedObject tr' (meta, LogValue "slotInEpoch" . PureI $ sNum `rem` epochSlots)
+                      traceNamedObject tr' (meta, LogValue "epoch" . PureI $ (sNum `div` epochSlots))
                   _ -> pure ()
               _ -> pure ()
+      where
+        slotN :: forall block . AF.HasHeader block
+              => AF.AnchoredFragment block -> Word64
+        slotN frag  = unSlotNo $ fromWithOrigin 0 (AF.headSlot frag)
+        -- Slot of the tip - slot @k@ blocks back. Use 0 as the slot for genesis
+        -- includes EBBs
+        slots :: forall block . AF.HasHeader block
+              => AF.AnchoredFragment block -> Word64
+        slots frag  = slotN frag
+                    - unSlotNo (fromWithOrigin 0 (pointSlot (AF.anchorPoint frag)))
```

Comment:
This is using the original suggestion not the updated one from https://github.com/input-output-hk/cardano-node/issues/250#issuecomment-545008126

The anchor point is the point _before_ the first block in the fragment. This is why it currently always reports one more slot. Use the version from Thomas's updated suggestion.

Reply:
done